man/rgbasm.5

```diff
@@ -324,9 +324,32 @@ with a zero constant as either operand will be constant 0, and
 .Sq ||
 with a non-zero constant as either operand will be constant 1, even if the other operand is non-constant.
 .Pp
-.Ic \&!
+.Sq \&!
 returns 1 if the operand was 0, and 0 otherwise.
 Even a non-constant operand with any non-zero bits will return 0.
+.Ss Integer functions
+Besides operators, there are also some functions which have more specialized uses.
+.EQ
+delim $$
+.EN
+.Bl -column "BITWIDTH(n)"
+.It Sy Name Ta Sy Operation
+.It Fn BITWIDTH n Ta Returns the number of bits necessary to represent
+.Ar n .
+Some useful formulas:
+.Ic BITWIDTH Ns Po Ar n Ns Pc - 1
+equals $\[lf] roman log2 ( n ) \[rf]$,
+.Ic BITWIDTH Ns Pq Ar n No - 1
+equals $\[lc] roman log2 ( n ) \[rc]$,
+and
+.Ic No 32 - Ic BITWIDTH Ns Pq Ar n
+equals $roman clz ( n )$.
+.It Fn TZCOUNT n Ta Returns $roman ctz ( n )$, the count of trailing zero bits at the end of the binary representation of
+.Ar n .
+.El
+.EQ
+delim off
+.EN
```

Comment:
This seems like a good candidate for merging with "Other functions". At least `HIGH` and `LOW` can be pulled from there.

Reply:
I wanted these close to "Operators", since they're all things that have numbers as input and output, and the distinction between operators and functions is minor. I considered whether to move `HIGH` and `LOW` into here, but decided to leave them grouped with the more complex "Other" functions which also deal with symbols, sections, etc, because they can act on registers as well as numbers.

Reply:
I'd argue that their use as registers counts less as “functions”, because they don't really manipulate *values* anymore. I think that use qualifies more as syntactical, especially because you *can't* use `HIGH(bc)` in an *expression*, which this is a sub-section of.

Thus, if both uses are specified in the same place, I think they should be split into separate sections—and `HIGH` and `LOW` moved with the new functions.

Reply:
Actually, I'd be fine with listing HIGH and LOW in both tables, with the numeric logic here and register logic explained in "Other".

Reply:
Not to me, because again, it's a syntactical feature and not part of the `EXPRESSIONS` section. I've found the descriptions confusing when I read them *a priori*, and they only make sense to me *a posteriori*.

I think I'd like the first `SYNTAX` section to contain a cross-ref to `gbz80(7)` (it's actually linked to at the beginning of `DESCRIPTION`, but I'd say the wording is vague and unclear), and then also a note that `HIGH(pair)` can be used as a substitute for `pa`.

Reply:
Okay, and then the descriptions of `HIGH` and `LOW` in the Functions table later wouldn't have to mention that register-syntactical detail.